### ISSUE TITLE
Issues with .prune / .clr

### ISSUE THREAD

**Issue by xnaas** [Kwoth/NadekoBot#834]
So I'm on the latest version of 1.0 and have noticed that `.prune` / `.clr` seems to be having some issues. On 0.99, if you ran `.prune` it would clear the bot's last 100 messages, but that doesn't happen anymore. If you specify an amount of messages to prune from a channel, then it'll prune some and still error out in the Command Prompt window.

![error!](http://i.imgur.com/7SrPjkU.png)

I *think* it's running into issues trying to prune its own messages, but I'm not 100% sure.

More screenies, basically the same thing: [same error](http://i.imgur.com/RlUacsn.png), [same error, again](http://i.imgur.com/1BrdRjL.png).

**Comment by fkndean** [comment]
I have made a fix for this, but doing dry run tests before making a PR in @Kwoth 's dnet.

**Comment by fkndean** [comment]
@Kwoth This has been fixed by you. Close this 👍

**Comment by xnaas** [comment]
Think you should re-open this, @Kwoth. If I try to prune a small number (ex: `.prune 4`), it's anyone's guess as to what messages it actually decides to delete. ¯\\\_(ツ)_/¯

The last time I ran it, it deleted 4 random messages I had sent recently. Not even just the last 4 I sent. 4 random ones. 😛

**Comment by xnaas** [comment]
Actually, after messing with it more, it's anything less than `.prune 7` that seems to not work right. Consistently. 😛

**Comment by xnaas** [comment]
Never mind. Did more testing. Still 100% fucked.

**Comment by Zylon** [comment]
I can't seem to re-create this behavior consistently, however it seems a lot more stable when just 1 person's messages are being removed. than when there are 15 people talking and then running .prune 50

It leads to some weird cases of it skipping a random amount of messages and hand-picking the messages it feels like.

tl;dr: still broken, and not sure why.

**Comment by xnaas** [comment]
@Kwoth was telling me it's an issue with Discord.NET, @Zylon. Basically, all we can do is wait for 100 Years™. 😛
